src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs

```diff
@@ -62,3 +62,34 @@ impl Rewriter for ArithExprSimplification {
         *expr = new;
     }
 }
+
+impl PlanRewriter for ArithExprSimplificationRule {
+    fn rewrite_logical_join(&mut self, join: &LogicalJoin) -> PlanRef {
+        let left = self.rewrite(join.left());
+        let right = self.rewrite(join.right());
+        Arc::new(join.clone_with_rewrite_expr(left, right, self))
+    }
```

Comment:
I don't think it's elegant to manually implement these functions for pure expression rules.
Is there a way to resolve that? cc @st1page

Reply:
maybe we could implement a kind of `PlanRewriter`, give it a Box<ExprRewriter> when initalizing it. all Expression rewrite demand will use the same code.

Reply:
or another way, how about doing expression simplificatation when construct the expression? so we can make sure any expression appearing in our project has been simplificated. just a half-baked idea.

Reply:
Anyway, we can refactor in future PRs.

Reply:
> or another way, how about doing expression simplificatation when construct the expression? so we can make sure any expression appearing in our project has been simplificated. just a half-baked idea.

This way we will lose the original expressions, not good for debugging. -1 (